File: Web/backend/app/models/course.py
"""
课程相关模型
"""

from datetime import datetime
from typing import Any, Optional, List
from pydantic import BaseModel


class CourseBase(BaseModel):
    """课程基础模型"""
    title: str
    slug: str
    description: Optional[str] = None
    cover_image: Optional[str] = None
    status: str = "draft"
    metadata: Optional[dict[str, Any]] = None


class Course(CourseBase):
    """课程完整模型"""
    id: str
    created_at: datetime
    updated_at: datetime
    created_by: Optional[str] = None


class CourseCreate(CourseBase):
    """创建课程"""
    pass


class CourseUpdate(BaseModel):
    """更新课程"""
    title: Optional[str] = None
    description: Optional[str] = None
    cover_image: Optional[str] = None
    status: Optional[str] = None
    metadata: Optional[dict[str, Any]] = None


class LessonBase(BaseModel):
    """课时基础模型"""
    slug: str
    title: str
    description: Optional[str] = None
    duration: Optional[str] = None
    order_index: int = 0
    metadata: Optional[dict[str, Any]] = None


class Lesson(LessonBase):
    """课时完整模型"""
    id: str
    course_id: str
    content: Optional[dict[str, Any]] = None
    content_source: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None


class LessonCreate(LessonBase):
    """创建课时"""
    course_id: str
    content: Optional[dict[str, Any]] = None
    content_source: Optional[str] = None


class LessonUpdate(BaseModel):
    """更新课时"""
    title: Optional[str] = None
    description: Optional[str] = None
    content: Optional[dict[str, Any]] = None
    content_source: Optional[str] = None
    duration: Optional[str] = None
    order_index: Optional[int] = None
    metadata: Optional[dict[str, Any]] = None


class QuestionBase(BaseModel):
    """题目基础模型"""
    type: str  # choice, multi_choice, fill_blank, drawing, essay, geogebra
    content: dict[str, Any]
    answer: Optional[dict[str, Any]] = None
    rubric: Optional[dict[str, Any]] = None
    points: int = 0
    order_index: int = 0
    metadata: Optional[dict[str, Any]] = None


class Question(QuestionBase):
    """题目完整模型"""
    id: str
    lesson_id: str
    created_at: datetime


class QuestionCreate(QuestionBase):
    """创建题目"""
    lesson_id: str


class QuestionUpdate(BaseModel):
    """更新题目"""
    content: Optional[dict[str, Any]] = None
    answer: Optional[dict[str, Any]] = None
    rubric: Optional[dict[str, Any]] = None
    points: Optional[int] = None
    order_index: Optional[int] = None
    metadata: Optional[dict[str, Any]] = None

File: Web/lib/schemas/course.ts
/**
 * 课程 / 课时领域 schema
 */
import { z } from 'zod';
import { UUIDSchema, DateTimeSchema } from './base';

export const CourseStatusSchema = z.enum(['draft', 'published', 'archived']);

export const CourseSchema = z.object({
  id: UUIDSchema,
  title: z.string(),
  slug: z.string(),
  description: z.string().nullable().optional(),
  cover_image: z.string().nullable().optional(),
  status: CourseStatusSchema.default('draft'),
  metadata: z.record(z.string(), z.any()).nullable().optional(),
  created_at: DateTimeSchema,
  updated_at: DateTimeSchema,
  created_by: UUIDSchema.nullable().optional(),
});

export type Course = z.infer<typeof CourseSchema>;

export const LessonSchema = z.object({
  id: UUIDSchema,
  course_id: UUIDSchema,
  slug: z.string(),
  title: z.string(),
  description: z.string().nullable().optional(),
  content: z.record(z.string(), z.any()).nullable().optional(),
  content_source: z.string().nullable().optional(),
  duration: z.string().nullable().optional(),
  order_index: z.number().default(0),
  metadata: z.record(z.string(), z.any()).nullable().optional(),
  created_at: DateTimeSchema,
  updated_at: DateTimeSchema.nullable().optional(),
});

export type Lesson = z.infer<typeof LessonSchema>;

/** 课程题目列表项（含课时信息，GET /api/v1/courses/[id]/questions） */
export interface CourseQuestion {
  id: string;
  lesson_id: string;
  lesson_slug: string;
  lesson_title: string;
  lesson_order_index?: number;
  type: string;
  stem: string;
  options: string[];
  points: number;
  order_index: number;
  correctIndex: number;
  explanation: string;
}
